### PR TITLE
Lint js, ts and vue files

### PR DESCRIPTION
**Pull request by micgro42**
This also sets up scripts to fix these files and automatic linting in a pre-commit hook.

Bug: T300889

### CODE REVIEW

**Comment by lucaswerkmeister** [comment]
With `root: true` in the eslint config (see new commit) it works.

**Comment by micgro42** [comment]
> With `root: true` in the eslint config (see new commit) it works.

Cool! +1 from my side to that extra commit

**Review by lucaswerkmeister — APPROVED** [review]
Do you know if it’s possible to move the ` --ext .js,.ts,.vue --ignore-path .gitignore` stuff to the eslintrc instead?

**Comment by micgro42** [comment]
> > Do you know if it’s possible to move the ` --ext .js,.ts,.vue --ignore-path .gitignore` stuff to the eslintrc instead?
> 
> I'll have a look before I merge this

Meh, that feature seems to have received a no-fix in eslint/eslint#3529 😕

File: .eslintrc.js [hunk 1]
```diff
@@ -0,0 +1,21 @@
...
+	root: true,
+	env: {
+		'vue/setup-compiler-macros': true,
+	},
+	extends: [
+		'wikimedia',
+		'wikimedia/node',
+		'wikimedia/client-common', // client-es6 still extends vue2 config
+		'wikimedia/vue3-es6',
+		'wikimedia/language/es2021', // this should ensure that we can use all the new featuresm, not only es6
```
Comment (lucaswerkmeister) [inline .eslintrc.js:11]:
```suggestion
		'wikimedia/language/es2021', // this should ensure that we can use all the new features, not only es6
```

File: src/App.vue [hunk 2]
```diff
@@ -1,7 +1,7 @@
 <script setup lang="ts">
 // This starter template is using Vue 3 <script setup> SFCs
 // Check out https://v3.vuejs.org/api/sfc-script-setup.html#sfc-script-setup
-import HelloWorld from './components/HelloWorld.vue'
+import HelloWorld from './components/HelloWorld.vue';
```
Comment (lucaswerkmeister) [inline src/App.vue:4]: Yup, I also noticed that this one was missing a semicolon in #9 :)

**Comment by micgro42** [comment]
This now overrides another rule to allow for the default order of these top-level tags in SFC: 
```
'vue/component-tags-order': [ 'error', { order: [ [ 'script', 'template' ], 'style' ] } ],
```

**Review by lucaswerkmeister — COMMENTED** [review]
Sorry, found one typo I must’ve missed earlier.

**Pull request merged**